### PR TITLE
LPS-109417 Simplify

### PR DESCRIPTION
**Pull request by dgarciasarai**
* b07669d1cd557b LPS-109417 Simplify (last commit)
* 80b2bc910dd071 LPS-109417 Fix typo
* a2c98801daebbe LPS-109417 Update tests
* 48ff2adf6c68bb LPS-109417 Calculate timeSpanComparator to display the reference dot and to filter the histogram
* 703394672f7ee2 LPS-109417 Build Lang
* f9bbb117b550a5 LPS-109417 Disable previous period button and add tooltip to inform the user
* aec484b28e6730 LPS-109417 Use default values from chartState
* d470bbfe2505bb LPS-109417 Disable previous period button if the publication date is in the actual time range
* d01ef7ed899c5a LPS-109417 Don't show data in chart before publication date
* 997345dda48abb LPS-109417 Extract
* 5c5c701602aeee LPS-109417 Fix example
* 24a9135a7786e3 LPS-109417 Add ReferenceDot for publish date in the chart and in the CustomTooltip
* 3bfc109de7bed8 LPS-109417 Remove unused method (first commit)

### CODE REVIEW

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:stable

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by andresfulla** [comment]
@dgarciasarai perhaps you having some **_git lag_**??!!

**Review by andresfulla — COMMENTED** [review]
Have in mind that whenever the histogram is empty, the `disabledPreviousPeriodButton` will be `undefined`. I think you want it to be a boolean

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.js [hunk 1]
```diff
@@ -272,14 +272,11 @@ export default function Chart({
...
 
 			const firstDate = new Date(firstDateLabel);
-			const lastDate = new Date(lastDateLabel);
 			const publishedDate = new Date(publishDate);
 
-			if (firstDate < publishedDate && lastDate > publishedDate) {
-				//Publish date between selected time range
+			if (firstDate < publishedDate) {
 				return true;
 			}
```
Comment (andresfulla) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.js]: Have in mind that whenever the histogram is empty, the disabledPreviousPeriodButton will be undefined. I think you want it to be a boolean

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.js [hunk 2]
```diff
@@ -220,10 +232,23 @@ export default function Chart({
 			}).then(data => {
 				if (!gone) {
 					if (isMounted()) {
+						let timeSpanComparator;
```
Comment (andresfulla) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.js]: [Opinion incoming] a ternary would be simpler in this case:
```
const timeComparator = chartState.tiimeSpanOption === LAST_24_HOURS ? HOUR_IN_MILLISECONDS : DAY_IN_MILLISECONDS
```

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.js [hunk 3]
```diff
@@ -242,19 +267,45 @@ export default function Chart({
...
+
+			const firstDate = new Date(firstDateLabel);
+			const publishedDate = new Date(publishDate);
+
+			if (firstDate < publishedDate) {
+				return true;
+			}
+
+			return false;
+		}
```
Comment (andresfulla) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.js:277]: bool!!

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.js [hunk 4]
```diff
@@ -391,7 +457,16 @@ export default function Chart({
 								separator={': '}
 							/>
 
-							{dataSet.keyList.map(keyName => {
+							<ReferenceDot
+								fill={'white'}
```
Comment (andresfulla) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.js]: Maybe put this params in constants, possibly on `CHART_SIZES`?

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/CustomTooltip.js [hunk 5]
```diff
@@ -11,19 +11,36 @@
...
 import PropTypes from 'prop-types';
 import React from 'react';
-
 /**
  * Component to customize the content of recharts Tooltip
  * http://recharts.org/en-US/api/Tooltip#content
  */
 export default function CustomTooltip(props) {
-	const {formatter, label, labelFormatter, payload, separator = ''} = props;
+	const {
```
Comment (andresfulla) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/CustomTooltip.js:19]: This component is weird. Published gets printed every time??

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/CustomTooltip.js [hunk 6]
```diff
@@ -11,19 +11,36 @@
...
+		fill,
```
Comment (andresfulla) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/CustomTooltip.js]: PropTypes!!

**Review by andresfulla — CHANGES_REQUESTED** [review]
I just gave you some feedback. The only thing that's not an opinion is the sometime-undefined boolean to disable the button, and the PropTypes update. 😍. Other than that: ssssspectacular!!!!

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/CustomTooltip.js (hunk 5 above)
Comment (dgarciasarai) [reply modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/CustomTooltip.js:19]: The publish date has to be displayed only in the information about that day.

**Comment by dgarciasarai** [comment]
https://github.com/andresfulla/liferay-portal/pull/223

**Pull request closed without merging**